pkg/envoy-controlplane/configcache.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package envoycontrolplane
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	envoyroutev3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	cachetypes "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	envoycachev3 "github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/google/uuid"
+	gocache "github.com/patrickmn/go-cache"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultCleanupInterval = 1 * time.Minute
+	NodeID                 = "kcp-ingress"
+)
+
+// configCache holds the current state of ingress resources and can generate Envoy snapshots.
+// Is uses a translator to do the Envoy resource translation.
+type configCache struct {
+	mu         sync.Mutex
+	ingresses  *gocache.Cache
+	translator *translator
+}
+
+func newConfigCache(translator *translator) *configCache {
+	return &configCache{
+		mu:         sync.Mutex{},
+		ingresses:  gocache.New(gocache.NoExpiration, defaultCleanupInterval),
```

Comment:
Do you anticipate using a nonzero expiration ever? If not, can we use a normal `map`?